app/models/block.rb

```diff
@@ -0,0 +1,42 @@
+class Block < ApplicationRecord
+  belongs_to :blocking_user, class_name: 'User', optional: false, inverse_of: :blocks
+  belongs_to :blocked_user, class_name: 'User', optional: false
+
+  validates :blocking_user_id, uniqueness: { scope: :blocked_user_id }
+  validates :hide_them, :hide_me, inclusion: { in: 0..2 }
+  validate :not_blocking_self
```

Comment:
Note to selves: need another validation to make sure at least one of hide them / hide me / interact is set